utils/fishing_handler.py

```diff
@@ -0,0 +1,25 @@
+from os import walk
+
+def fetch_fish(dir:str):
+    """Fetch fish given a directory."""
+    fetched_fish = {
+        "common": {}, 
+        "uncommon": {}, 
+        "rare": {},
+        "epic": {},
+        "legendary": {},
+        "mythic": {},
+    }
+    
+    _, _, filenames = next(walk(dir))
+    for i in filenames:
+        splitted = i.split("_")
+        try:
+            fetched_fish[splitted[0]][" ".join(splitted[2:])[:-4]] = {
+                "multiplier": splitted[1],
+                "name": " ".join(splitted[2:])[:-4].title(),
+            }
+        except KeyError:
+            pass
+        final:
+            return fetched_fish
```

Comment:
Shouldn't this return be outside of your loop? Otherwise we'll only get one fish in the dictionary, surely.

Reply:
..isn't it already in a loop?

Reply:
Yes. The return is inside the loop. So it returns before it's looped through all the fish.